### PR TITLE
feat: sso setup

### PR DESCRIPTION
**Pull request by robot9706**
Made all SSO settings optional to only init providers which have their client settings.

Modified `SELFSERVICE_METHODS_OIDC_CONFIG_PROVIDERS_0_CLIENT_ID` like variables to reflect their provider: `OIDC_GILAB_CLIENT_ID`, `OIDC_GILAB_CLIENT_SECRET`, etc.

### CODE REVIEW

File: web/kratos/Dockerfile [hunk 1]
```diff
@@ -7,6 +7,7 @@ RUN apk add gettext
 COPY . /usr/bin/app/kratos
 
 RUN mkdir -p /etc/config/kratos && chmod a+x /usr/bin/app/kratos/entrypoint.sh && \
+    chmod a+x /usr/bin/app/kratos/oidc-setup.sh && \
```
Comment (polaroi8d) [inline web/kratos/Dockerfile:10]: The `a` is needed?

File: web/kratos/oidc-setup.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env sh
```
Comment (polaroi8d) [inline web/kratos/oidc-setup.sh:1]: It would be better to create a function that handles repetitive logic for checking environment variables, IMHO.

**Review by polaroi8d — CHANGES_REQUESTED** [review]

File: web/kratos/Dockerfile (hunk 1 above)
Comment (robot9706) [reply web/kratos/Dockerfile:10]: Copied from the line above so they are the same.

File: web/kratos/oidc-setup.sh (hunk 2 above)
Comment (robot9706) [reply web/kratos/oidc-setup.sh:1]: Okay I'll try to do it.

**Review by polaroi8d — APPROVED** [review]

**Pull request merged**